File: backend/app/routes/review.py
from flask import Blueprint, jsonify, request
from flask_jwt_extended import jwt_required, get_jwt_identity
from app.models.review_plan import ReviewPlan
from app.models.word_practice import WordPractice
from app import db
from datetime import datetime

bp = Blueprint('review', __name__, url_prefix='/api/review')

@bp.route('/plans', methods=['GET'])
@jwt_required()
def get_review_plans():
    user_id = get_jwt_identity()
    plans = ReviewPlan.query.filter_by(user_id=user_id).all()
    return jsonify([plan.to_dict() for plan in plans])

@bp.route('/stats', methods=['GET'])
@jwt_required()
def get_review_stats():
    user_id = get_jwt_identity()
    total = ReviewPlan.query.filter_by(user_id=user_id).count()
    completed = ReviewPlan.query.filter_by(user_id=user_id, status='completed').count()
    pending = total - completed
    return jsonify({
        'total': total,
        'completed': completed,
        'pending': pending
    })

@bp.route('/<int:plan_id>/start', methods=['POST'])
@jwt_required()
def start_review(plan_id):
    user_id = get_jwt_identity()
    plan = ReviewPlan.query.filter_by(id=plan_id, user_id=user_id).first_or_404()
    return jsonify(plan.to_dict())

@bp.route('/<int:plan_id>/complete', methods=['POST'])
@jwt_required()
def complete_review(plan_id):
    user_id = get_jwt_identity()
    plan = ReviewPlan.query.filter_by(id=plan_id, user_id=user_id).first_or_404()
    
    data = request.get_json()
    is_correct = data.get('isCorrect', False)
    
    # 更新复习计划状态
    plan.status = 'completed'
    plan.last_review_time = datetime.utcnow()
    
    # 更新单词练习记录
    practice = WordPractice(
        user_id=user_id,
        word_id=plan.word_id,
        is_correct=is_correct,
        review_time=datetime.utcnow()
    )
    db.session.add(practice)
    db.session.commit()
    
    return jsonify({'message': '复习完成'}) 

File: frontend/src/services/review.ts
import { request } from './request';

export interface ReviewPlan {
  id: number;
  title: string;
  description: string;
  startTime: string;
  endTime: string;
  status: 'pending' | 'completed' | 'overdue';
  userId: number;
  createdAt: string;
  updatedAt: string;
}

export interface ReviewStats {
  total: number;
  completed: number;
  pending: number;
  completionRate: number;
  averageScore: number;
  dailyStats: Array<{
    date: string;
    completed: number;
    total: number;
  }>;
}

export const reviewService = {
  getReviewPlans: async (): Promise<ReviewPlan[]> => {
    const response = await request.get('/api/review/plans');
    return response.data;
  },

  getReviewStats: async (): Promise<ReviewStats> => {
    const response = await request.get('/api/review/stats');
    return response.data;
  },

  createReviewPlan: async (data: {
    title: string;
    description: string;
    reviewTime: [string, string];
  }): Promise<ReviewPlan> => {
    const [startTime, endTime] = data.reviewTime;
    const response = await request.post('/api/review/plans', {
      title: data.title,
      description: data.description,
      startTime,
      endTime,
    });
    return response.data;
  },

  completeReview: async (id: number): Promise<void> => {
    await request.post(`/api/review/plans/${id}/complete`);
  },

  deleteReviewPlan: async (id: number): Promise<void> => {
    await request.delete(`/api/review/plans/${id}`);
  },

  updateReviewPlan: async (
    id: number,
    data: Partial<ReviewPlan>
  ): Promise<ReviewPlan> => {
    const response = await request.put(`/api/review/plans/${id}`, data);
    return response.data;
  },
}; 